File: app/src/main/java/br/com/victorlsn/mytaxi/interfaces/CarListMVP.java
package br.com.victorlsn.mytaxi.interfaces;

import android.content.Context;

import java.util.List;
import java.util.Map;

import br.com.victorlsn.mytaxi.beans.Car;

/**
 * Created by victorlsn on 26/02/19.
 *
 */

public interface CarListMVP {
    interface Model {
        void getVehicles(Context context, Map<String, String> coordinates);
    }

    interface Presenter extends BaseMVP.Presenter {
        void requestVehicles(Context context);

        void requestVehiclesSuccessfully(List<Car> cars);

        void requestVehiclesFailure(String error);
    }

    interface View extends BaseMVP.View {
        void receiveVehiclesList(List<Car> cars);
    }
}

File: app/src/main/java/br/com/victorlsn/mytaxi/interfaces/CarListMVP.kt
package br.com.victorlsn.mytaxi.interfaces

import android.content.Context

import br.com.victorlsn.mytaxi.beans.Car

/**
 * Created by victorlsn on 26/02/19.
 *
 */

interface CarListMVP {
    interface Model {
        fun getVehicles(context: Context, coordinates: Map<String, String>)
    }

    interface Presenter : BaseMVP.Presenter {
        fun requestVehicles(context: Context)

        fun requestVehiclesSuccessfully(cars: List<Car>)

        fun requestVehiclesFailure(error: String?)
    }

    interface View : BaseMVP.View {
        fun receiveVehiclesList(cars: List<Car>?)
    }
}
